test/e2e/suites/e2e/self_hosted_test.go

```diff
@@ -0,0 +1,43 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+
+	"sigs.k8s.io/cluster-api-provider-openstack/test/e2e/shared"
+)
+
+var _ = Describe("When testing Cluster API provider Openstack working on [self-hosted] clusters", func() {
```

Comment:
```suggestion
var _ = Describe("Create a Cluster API Provider OpenStack cluster, pivote to a self-hosted cluster", func() {
```

Reply:
This is the style used in CAPI and also in the [clusterctl upgrade test in CAPO](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/576a13f19f5fc52a063c039d6734b72fb3d62ecd/test/e2e/suites/e2e/clusterctl_upgrade_test.go#L36). I'm open to change it but would like to explain why it is expressed in this way first.
The top level `Describe` text is followed by `It` and `By` blocks in the test, so this would read something like

```
[Describe] When testing Cluster API provider Openstack working on [self-hosted] clusters
[It] Should pivot the bootstrap cluster to a self-hosted cluster
[By] Creating a workload cluster
...
```